src/api/nodeRpcCall.js

```diff
@@ -90,45 +95,47 @@ const sendTransferLLM = async (payload, callback) => {
   const transferExtrinsic = api.tx.llm.sendLlm(account_to, (meritsToGrains(amount)));
 
   const injector = await web3FromSource('polkadot-js');
-  transferExtrinsic.signAndSend(account_from, { signer: injector.signer }, ({ status }) => {
+  transferExtrinsic.signAndSend(account_from, { signer: injector.signer }, ({ status, events, dispatchError }) => {
+    let errorData = handleMyDispatchErrors(dispatchError, api)
     if (status.isInBlock) {
       // eslint-disable-next-line no-console
       console.log(`Completed at block hash #${status.asInBlock.toString()}`);
-      callback(null, status.asInBlock.toString());
+      callback(null, {
+        blockHash: status.asInBlock.toString(),
+        errorData
+      });
     }
   }).catch((error) => {
     // eslint-disable-next-line no-console
     console.error(':( transaction failed', error);
-    callback(error);
+    callback({isError: true, details: error.toString()});
   });
 };
 
 const stakeToPolkaBondAndExtra = async (payload, callback) => {
-  try {
-    const { values: { amount }, isUserHaveStake, walletAddress } = payload;
-    const api = await getApi();
-    const transferExtrinsic = isUserHaveStake
-      ? await api.tx.staking.bondExtra(`${amount}000000000000`)
-      : await api.tx.staking.bond(walletAddress, `${amount}000000000000`, 'Staked');
-
-    const injector = await web3FromSource('polkadot-js');
-    // eslint-disable-next-line max-len
-    await transferExtrinsic.signAndSend(walletAddress, { signer: injector.signer }, ({ status }) => {
-      if (status.isInBlock) {
-        // eslint-disable-next-line no-console
-        console.log(`InBlock at block hash #${status.asInBlock.toString()}`);
-        callback(null, 'done');
-      }
-    }).catch((error) => {
+  const { values: { amount }, isUserHaveStake, walletAddress } = payload;
+  const api = await getApi();
+  const transferExtrinsic = isUserHaveStake
+    ? await api.tx.staking.bondExtra(`${amount}000000000000`)
```

Comment:
How about using `dollarsToGrains` from `walletHelpers.js`?
This looks risky - what if amount is "15.0"?

Reply:
Additionally, although this is subjective, so treat this as optional - converting to grains should probably be done ASAP at the input, not deep down in nodeRpcCall.js.